### ISSUE TITLE
Equivalence of gradient accumulation

### ISSUE THREAD

**Issue by TonyLianLong** [microsoft/Swin-Transformer#107]
Thanks for the amazing project. I have a question about the equivalence of enabling/not enabling gradient accumulation.

Since there is no inter-sample communication such as BatchNorm, a config of `--batch-size 128` is exactly equivalent to `--batch-size 64 --accumulation-steps 2` in terms of the gradient's numerical value and thus the optimized model. Am I right on this?

**Comment by ancientmooner** [comment]
Yes, you are correct.

**Comment by lawpdas** [comment]
Hi @TonyLianLong 
Have you got the correct result?
I use the swin backbone in a downstream task and use gradient accumulation in my code, but it cannot perform as good as ResNet. Could you give me some advice?
